lib/Mail/Milter/Authentication/Protocol/SMTP.pm

```diff
@@ -414,10 +414,14 @@ sub smtp_command_mailfrom {
         $ip = substr( $ip, 5 );
     }
 
-    $self->logdebug( "Inbound IP Address $ip" );
-    $returncode = $handler->top_connect_callback( $host, Net::IP->new( $ip ) );
+    # Do connection remapping first
+    $handler->remap_connect_callback( $host, Net::IP->new( $ip ) );
+    $handler->remap_helo_callback( $helo );
```

Comment:
Is there a reason to do this inside of smtp_command_mailfrom instead of smtp_init? One is called right after the other in protocol_process_request. Since we're remapping IPs and HELO names, we should already have those by the time we call smtp_init, right?

Reply:
smtp_init sets up some initial state, and is called in a few other places. inside the mailfrom method seemed the most sensible as it's also where the relevant top callbacks are made.